guide/message_transformation/message_transformation.bal

```diff
@@ -0,0 +1,370 @@
+                    res.statusCode = 400;
+                    res.setPayload(untaint err.message);
+                    caller->respond(res) but { error e =>
+                    log:printError("Error sending response", err = e) };
+                }else if (!check msg.gender.toString().matches("[a-zA-Z]+")) {
+                    error err = { message: "student gender containts invalid data" };
+                    http:Response res = new;
+                    res.statusCode = 400;
+                    res.setPayload(untaint err.message);
+                    caller->respond(res) but { error e =>
+                    log:printError("Error sending response", err = e) };
+                }
+                else {
+                    //Assign user input values to variables
+                    int Idvalue = check <int>msg["id"];
+                    string nameString = check <string>msg["name"];
+                    string cityString = check <string>msg["city"];
+                    string genderString = check <string>msg["gender"];
+                    //add values to the student details table
+                    ret = testDB->update("INSERT INTO student(id, name, city, gender) values (?, ?, ?, ?)", Idvalue,
+                        nameString, cityString, genderString);
+                    handleUpdate(ret, "Add details to the table");
+                    json iddetails= { id: Idvalue };
+                    //set filtered payload to the request
+                    filteredreq.setJsonPayload(untaint iddetails);
+                    //forward request to the nesxt ID validating service
+                    var clientResponse = validatorEP->forward("/", filteredreq);
+                    match clientResponse {
+                        http:Response res => {
+                            caller->respond(res) but { error e =>
+                            log:printError("Error sending response", err = e) };
+                        }
+                        error err => {
+                            http:Response res = new;
+                            res.statusCode = 500;
+                            res.setPayload(err.message);
+                            caller->respond(res) but { error e =>
+                            log:printError("Error sending response", err = e) };
+                        }
+                    }
+                }
+            }
+            error err => {
+                http:Response res = new;
+                res.statusCode = 500;
+                res.setPayload( untaint err.message);
+                caller->respond(res) but { error e =>
+                log:printError("Error while content reading", err = e) };
+            }
+        }
+
+    }
+}
+
+//the student ID validator service
+service<http:Service> validator bind claimvaditadeEP {
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/"
+    }
+    validator(endpoint caller, http:Request filteredreq) {
```

Comment:
Better if you can rename resource name to "validate"

Reply:
Done